### PR TITLE
nodejs*: normalise names to better fit other packages

### PR DESCRIPTION
**Pull request by SuperSandro2000**
Versioned package attributes are usually named like $pname_$version

ToDo after merge: send a PR to node2nix.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fricklerhandwerk — APPROVED** [review]

**Review by winterqt — CHANGES_REQUESTED** [review]
1. Please fix eval.
2. This seems to add a `node-composition.nix` file to `open-stage-control`, when it is not needed (the package uses `buildNpmPackage`).

**Comment by SuperSandro2000** [comment]
@ofborg eval

FYI @winterqt

**Comment by teto** [comment]
is there an RFC for this ? I wonder about doing something similar to the lua ecosystem but I felt the churn would hurt more than the lack of consistency, especially if there is no written down rule about this.

**Comment by SuperSandro2000** [comment]
> is there an RFC for this ? I wonder about doing something similar to the lua ecosystem but I felt the churn would hurt more than the lack of consistency, especially if there is no written down rule about this.

It follows other packages which do similar things. We don't need an RFC for this.

**Comment by fricklerhandwerk** [comment]
@infinisil do you want to take a look at this? The PR is constantly diverging, I think we should make a decision soon.

**Comment by SuperSandro2000** [comment]
> However, it's really hard to see what actually changed due to the size of this PR and there being no separation into commits.

I disagree with you. The total commit diff is just over 200 lines.

The relevant part is

```diff
@@ -9377,32 +9377,32 @@ with pkgs;
 
   nodenv = callPackage ../development/tools/nodenv { };
 
-  nodejs = hiPrio nodejs-18_x;
+  nodejs = hiPrio nodejs_18;
 
-  nodejs-slim = nodejs-slim-18_x;
+  nodejs-slim = nodejs-slim_18;
 
-  nodejs-14_x = callPackage ../development/web/nodejs/v14.nix {
+  nodejs_14 = callPackage ../development/web/nodejs/v14.nix {
     openssl = openssl_1_1;
   };
-  nodejs-slim-14_x = callPackage ../development/web/nodejs/v14.nix {
+  nodejs-slim_14 = callPackage ../development/web/nodejs/v14.nix {
     openssl = openssl_1_1;
     enableNpm = false;
   };
-  nodejs-16_x = callPackage ../development/web/nodejs/v16.nix { };
-  nodejs-slim-16_x = callPackage ../development/web/nodejs/v16.nix {
+  nodejs_16 = callPackage ../development/web/nodejs/v16.nix { };
+  nodejs-slim_16 = callPackage ../development/web/nodejs/v16.nix {
     enableNpm = false;
   };
-  nodejs-18_x = callPackage ../development/web/nodejs/v18.nix { };
-  nodejs-slim-18_x = callPackage ../development/web/nodejs/v18.nix {
+  nodejs_18 = callPackage ../development/web/nodejs/v18.nix { };
+  nodejs-slim_18 = callPackage ../development/web/nodejs/v18.nix {
     enableNpm = false;
   };
-  nodejs-19_x = callPackage ../development/web/nodejs/v19.nix { };
-  nodejs-slim-19_x = callPackage ../development/web/nodejs/v19.nix {
+  nodejs_19 = callPackage ../development/web/nodejs/v19.nix { };
+  nodejs-slim_19 = callPackage ../development/web/nodejs/v19.nix {
     enableNpm = false;
   };
   # Update this when adding the newest nodejs major version!
-  nodejs_latest = nodejs-19_x;
-  nodejs-slim_latest = nodejs-slim-19_x;
+  nodejs_latest = nodejs_19;
+  nodejs-slim_latest = nodejs-slim_19;
 
   buildNpmPackage = callPackage ../build-support/node/build-npm-package { };
```

So basically ``nodejs-14_x`` -> ``nodejs_14``

> I think it should be a first commit for the renames from `nodejs_<n>_x` to `nodejs_<n>`, and then another that does the treewide change

This creates a commit where you cannot bisect on because there a many evaluations errors.

**Comment by SuperSandro2000** [comment]
Going to merge this now to now have to rebase it another time. The new nodejs_20 is already named like the new style and having this inconsistent in 23.05 is not doing us a favor.

The commit message is good enough for me, I improved the wording with one of the last force pushed a bit and IMO discussing if that commit should be split or not has not a big enough impact that it is worth our time. It can be displayed in the github web UI unlike many other treewide commits which often almost crash your browser.

**Comment by infinisil** [comment]
> This creates a commit where you cannot bisect on because there a many evaluations errors.

Then flip the order around: An initial commit to introduce the aliases in `all-packages.nix`, another commit to do the treewide change, and another to move the aliases to `aliases.nix`. Generally there's always a decent way to order commits, please consider doing this in the future to make reviews easier.

And also generally, if you get feedback to a PR, at least wait until the feedback is resolved if you're gonna merge it yourself. But even better, don't merge your own PR's, only do that in justified situations.

**Comment by SuperSandro2000** [comment]
I just thought about it and I know from my slightly chaotic working style that I will definitely not put everything into the correct commit first time and so some slight mistakes. For bigger, treewide changes that might create some more clarity but for this kind of scope I think that me taking multiple tries to get it right and the less straightforward history outway the slightly bigger commit.

**Comment by winterqt** [comment]
> And also generally, if you get feedback to a PR, at least wait until the feedback is resolved if you're gonna merge it yourself. But even better, don't merge your own PR's, only do that in justified situations.

I'm going to, per usual, emphasize this. I actually don't even have much of an issue with the change, but self-merging things that people are actively looking at it *absurdly irritating*. I have said this to you personally so, so many times: **please stop it**.

This is part of the reason why I've stepped away from the project for a bit, to maintain my sanity with regards to things like this. 😕

File: pkgs/development/node-packages/overrides.nix
Comment (lilyinstarlight) [inline pkgs/development/node-packages/overrides.nix:360]: @SuperSandro2000, was this second patch to node2nix removed intentionally or in error?

It is absolutely still needed

Comment (SuperSandro2000) [reply pkgs/development/node-packages/overrides.nix:360]: Yeah, that was an error from me. #228545

**Comment by jonringer** [comment]
> I'm going to, per usual, emphasize this. I actually don't even have much of an issue with the change, but self-merging things that people are actively looking at it absurdly irritating.

Not condoning the actions in this PR, but having PRs open for months is also irritating. Having an unbounded timeline for getting things merged really kills a lot of motivation to do certain things.

Nixpkgs really needs more ways to automate the "fluff" workload (e.g. small version bumps), so that it doesn't just seem like a constant stream of "work debt" for reviewers.

**Comment by winterqt** [comment]
> Not condoning the actions in this PR, but having PRs open for months is also irritating.

I get it, but the PR was literally reviewed by someone else one day before some pushes + the self-merge. That is not fair, in my opinion.

**Pull request merged**
